mathesar/models.py

```diff
@@ -211,6 +215,66 @@ def update_record(self, id_value, record_data):
     def delete_record(self, id_value):
         return records.delete_record(self._sa_table, self.schema._sa_engine, id_value)
 
+    def get_constraint_by_type_and_columns(self, constraint_type, columns):
+        for constraint in self.sa_constraints:
+            current_constraint_type = constraints.get_constraint_type_from_class(constraint)
+            if current_constraint_type == constraint_type and set(columns) == set([column.name for column in constraint.columns]):
+                return constraint
+        return None
+
+    def add_constraint(self, constraint_data):
+        if 'type' not in constraint_data or constraint_data['type'] != constraints.ConstraintType.UNIQUE.value:
+            raise ValueError('Only creating unique constraints is currently supported.')
+        constraints.create_unique_constraint(
+            self._sa_table.name,
+            self._sa_table.schema,
+            self.schema._sa_engine,
+            constraint_data['columns'],
+            constraint_data['name'] if 'name' in constraint_data else None
+        )
+        try:
+            # Clearing cache so that new constraint shows up.
+            del self._sa_table
+        except AttributeError:
+            pass
+        sa_constraint = self.get_constraint_by_type_and_columns(
+            constraint_data['type'],
+            constraint_data['columns']
+        )
+        engine = self.schema.database._sa_engine
+        constraint_oid = constraints.get_constraint_oid_by_name_and_table_oid(sa_constraint.name, self.oid, engine)
+        return Constraint.objects.create(oid=constraint_oid, table=self)
```

Comment:
You can have more than one constraint on the same columns of the same type.  This might happen through migrations, user inattention, etc.  This could end up reflecting the wrong one, and running into confusion when trying to add a Constraint object with a previously-reflected OID, rather than the one intended.

Reply:
This is fixed. Since we know the pattern of the name, I generated the name instead of trying to get the constraint by type/columns. The name should be unique.